internal/elements.go

```diff
@@ -417,6 +418,15 @@ type CurrentUserPrincipal struct {
 	Unauthenticated *struct{} `xml:"unauthenticated,omitempty"`
 }
 
+// https://datatracker.ietf.org/doc/html/rfc3744#section-5.4
+type CurrentUserPrivilegeSet struct {
+	XMLName    xml.Name    `xml:"DAV: current-user-privilege-set"`
+	Privileges []Privilege `xml:"privilege>dynamic"`
```

Comment:
Nit: this file use the same name as the XML element even if there may be multiple.